File: Searching Algorithms/linear_search.py
# function returning if the target is present in the array or not.
def linear_search(arr, target):
    for a in arr:
        if a==target:
            return True
    return False

arr = [4,7,3,5,6,1,8,2,9]
print(linear_search(arr, 5))
print(linear_search(arr, 10))

# function returning the index of the target in the array, if present (otherwise returns -1)
def linear_search_index(arr, target):
    index = -1
    for i, a in enumerate(arr):
        if a==target:
            index = i
            return index
    return index

print(linear_search_index(arr, 5))
print(linear_search_index(arr, 10))

File: Searching Algorithms/linear_search.cpp
#include <iostream>
#include <vector>
using namespace std;

int linear_search(vector<int> &nums, int target){
    int index = -1;
    int n = nums.size();

    for(int i = 0; i<n; i++){
        // cout<<nums[i]<<" ";
        if (nums[i]==target){
            index = i;
            return index;
        }
    }
    return index;
}

int main(){
    vector<int> nums = {4, 7, 3, 5, 6, 1, 8, 2, 9};
    cout<<linear_search(nums, 5);
    return 0;
}